### PR TITLE
Supporting the column compression option on MariaDB

### LINKED ISSUE #14956: Implement Page/Column Compression options
Make use of the new PAGE_COMPRESSED=1 and Column's COMPRESSED=1 features.

Page compression: https://mariadb.com/kb/en/library/compression/

Column compression: https://jira.mariadb.org/browse/MDEV-11371

These don't seem to be available anywhere on phpMyAdmin.

(edited post as I was also reporting the issue on #14673, which appears to have a different cause)

### PR DESCRIPTION
**Pull request by iifawzi**
Hi, this PR should fix #14956
Signed-off-by: Fawzi E. Abdulfattah <iifawzie@gmail.com>
Fixes #14956

image

### CODE REVIEW

**Comment by williamdes** [comment]
@MauricioFauth could this go to 5.2 ?

File: libraries/classes/Types.php [hunk 1]
```diff
@@ -713,13 +713,22 @@ public function getAllFunctions()
...
+
+        $attributes = [
             '',
             'BINARY',
             'UNSIGNED',
             'UNSIGNED ZEROFILL',
             'on update CURRENT_TIMESTAMP',
         ];
+
+        if ($isMariaDB && $serverVersion >= 100100) {
```
Comment (iifawzi) [inline libraries/classes/Types.php:727]: I've assumed that it's supported starting from `10.1.0`, based on this 
https://mariadb.com/kb/en/innodb-page-compression/#comment_1992

File: libraries/classes/Util.php [hunk 2]
```diff
@@ -1357,6 +1360,10 @@ public static function extractColumnSpec($columnSpecification)
             $attribute = 'UNSIGNED ZEROFILL';
         }
 
+        if ($compressed) {
+            $attribute = 'COMPRESSED=zlib';
```
Comment (williamdes) [inline libraries/classes/Util.php:1364]: it's not always zlib ?
Can we use what was parsed ?

Comment (iifawzi) [reply libraries/classes/Util.php:1364]: In fact, from this particular scope, no. we're indicating that the compression is used from the `compressed` word shown in the type, which's the way we also use to detect other attrs, `BINARY`, `UNSIGNED`, or`UNSIGNED ZEROFILL`. getting it dynamically from the parser would need more work

Comment (williamdes) [reply libraries/classes/Util.php:1364]: Okay, thanks for clarifying

**Review by williamdes — APPROVED** [review]

**Comment by linaspasv** [comment]
What is the state of this? The support for MariaDB is very needed.

**Review by MauricioFauth — APPROVED** [review]

**Comment by williamdes** [comment]
For the record:
> With InnoDB page compression, multiple compression algorithms are supported. In contrast, with InnoDB's [COMPRESSED](https://mariadb.com/kb/en/innodb-compressed-row-format/) row format, [zlib](https://www.zlib.net/) is the only supported compression algorithm. This means that the [COMPRESSED](https://mariadb.com/kb/en/innodb-compressed-row-format/) row format has less compression options than InnoDB page compression does.

Ref: https://mariadb.com/kb/en/innodb-page-compression/#comparison-with-the-compressed-row-format

**Pull request merged**